src/analysis/sensitivity.jl

```diff
@@ -401,18 +402,45 @@ ADRIA.sensitivity.rsa(X, y; S=10)
    https://dx.doi.org/10.1002/9780470725184
    Accessible at: http://www.andreasaltelli.eu/file/repository/Primer_Corrected_2022.pdf
 """
-function rsa(X::DataFrame, y::AbstractVector{<:Real}; S::Int64=10)::NamedDimsArray
+function rsa(
+    X::DataFrame, y::AbstractVector{<:Real}, model_spec::DataFrame; S::Int64=10
+)::NamedDimsArray
     N, D = size(X)
-    seq = collect(0.0:(1/S):1.0)
 
     X_di = @MVector zeros(N)
+    sel = trues(N)
+    factors = Symbol.(names(X))
+
+    factors_to_assess = model_spec.fieldname .∈ [factors]
+    foi_attributes = Symbol[:fieldname, :ptype, :lower_bound, :upper_bound]
+    foi_spec = model_spec[factors_to_assess, foi_attributes]
```

Comment:
This doesn't work because the factor names in the dataframe are strings, but `factors` are cast to Symbols.

Have you checked if this works, Rose?

Reply:
Hi Takuya, yes I did check. I found that the factors from the domain model spec are symbols but the rs model spec are strings - is this intended? Or do I need to rebase again. I do check my work

Reply:
> Hi Takuya, yes I did check. I found that the factors from the domain model spec are symbols but the rs model spec are strings - is this intended? Or do I need to rebase again. I do check my work

Okay, then it's a mismatch between how `load_results()` reads in the stored copy of the model specification, and how `Domain` treats it. 

Remember that the `ResultSet` is intended to be a standalone snapshot of what was used to run scenarios. You will not always have the `Domain` available to generate the model spec from.

Reply:
Ok I will change the usage.

Reply:
No I think there's a misunderstanding here. We have to support both.

Reply:
ah ok, sorry I thought you meant you wanted to support the result set one only